app/src/main/java/org/mifos/mobilebanking/utils/gcm/MyGcmListenerService.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mifos.mobilebanking.utils.gcm;
+
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.content.LocalBroadcastManager;
+import android.util.Log;
+
+import com.google.android.gms.gcm.GcmListenerService;
+
+import org.mifos.mobilebanking.R;
+import org.mifos.mobilebanking.models.notification.NotificationMifosMobile;
+import org.mifos.mobilebanking.ui.activities.HomeActivity;
+import org.mifos.mobilebanking.utils.Constants;
+
+public class MyGcmListenerService extends GcmListenerService {
+
+    private static final String TAG = "MyGcmListenerService";
+
+    /**
+     * Called when message is received.
+     *
+     * @param from SenderID of the sender.
+     * @param data Data bundle containing message data as key/value pairs.
+     *             For Set of keys use data.keySet().
+     */
+    // [START receive_message]
+    @Override
+    public void onMessageReceived(String from, Bundle data) {
+        String message = data.getString("message");
+        Log.d(TAG, "From: " + from);
+        Log.d(TAG, "Message: " + message);
+
+        if (from.startsWith("/topics/")) {
+            // message received from some topic.
+        } else {
+            // normal downstream message.
+        }
+
+        sendNotification(message);
+        Intent registrationComplete = new Intent(Constants.NOTIFY_HOME_FRAGMENT);
+        LocalBroadcastManager.getInstance(this).sendBroadcast(registrationComplete);
+    }
+    // [END receive_message]
+
+    /**
+     * Create and show a simple notification containing the received GCM message.
+     *
+     * @param message GCM message received.
+     */
+    private void sendNotification(String message) {
+        Intent intent = new Intent(this, HomeActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        intent.putExtra("notification", true);
```

Comment:
Please don't use hard-coded string, It creates misunderstanding and It's hard to debug sometime.